modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java

```diff
@@ -143,107 +122,266 @@ private Device generateDevice(String type, String deviceID, String osVersion, St
      */
     @Override
     public Response getResponse(Document request)
-            throws WindowsDeviceEnrolmentException, WindowsOperationException {
+            throws WindowsDeviceEnrolmentException, WindowsOperationException, NotificationManagementException,
+            WindowsConfigurationException {
 
-        String val = SyncmlServiceImpl.getStringFromDoc(request);
-        int msgID;
+        int msgId;
         int sessionId;
         String user;
         String token;
         String response;
         SyncmlDocument syncmlDocument;
-        List<Operation> deviceInfoList;
+        List<Operation> deviceInfoOperations;
         List<? extends Operation> pendingOperations;
+        OperationUtils operationUtils = new OperationUtils();
+        DeviceInfo deviceInfo = new DeviceInfo();
 
-        syncmlDocument = SyncmlParser.parseSyncmlPayload(request);
-        sessionId = syncmlDocument.getHeader().getSessionId();
-        user = syncmlDocument.getHeader().getSource().getLocName();
-        DeviceIdentifier deviceIdentifier = convertToDeviceIdentifierObject(syncmlDocument.getHeader().getSource()
-                                                                                    .getLocURI());
-        msgID = syncmlDocument.getHeader().getMsgID();
-
-        if (SYNCML_FIRST_MESSAGE == msgID && SESSIONID_FIRST == sessionId) {
-            token = syncmlDocument.getHeader().getCredential().getData();
-            CacheEntry cacheToken = (CacheEntry) DeviceUtil.getCacheEntry(token);
-            if (cacheToken.getUsername().equals(user)) {
-
-                if (enrollDevice(request)) {
-                    deviceInfoList = getDeviceInfo();
-                    response = generateReply(syncmlDocument, deviceInfoList);
-                    return Response.ok().entity(response).build();
-                } else {
-                    String msg = "Error occurred in device enrollment.";
-                    log.error(msg);
-                    return Response.status(500).entity(msg).build();
+        try {
+            if (SyncmlParser.parseSyncmlPayload(request) != null) {
+                try {
+                    syncmlDocument = SyncmlParser.parseSyncmlPayload(request);
+                } catch (SyncmlMessageFormatException e) {
+                    String msg = "Error occurred due to bad syncml format";
+                    log.error(msg, e);
```

Comment:
yes, as this is exposed externally, but should not do any logging in private methods used by this.